### PR TITLE
Zope 4 fixes.

### PR DESCRIPTION
**Pull request by jensens**
Fixes made in order to work with Zope 4, includes some common findings.

### CODE REVIEW

**Comment by icemac** [comment]
It would be nice to see if these changes pass on Travis, see #5 which still requires Travis CI to be switched on for this repository (#6).

**Comment by icemac** [comment]
Travis is enabled now, could you please rebase your PR to the current master?

**Comment by jensens** [comment]
rebased

tests are red, seems like a returned 302 signature and actual headers are different (so not wrong afaict).

File: setup.py [hunk 1]
```diff
@@ -50,6 +50,7 @@ def _package_doc(name):
           'Products.PluginRegistry',
           'Products.GenericSetup',
           'Products.StandardCacheManagers',
+          'ZServer',
```
Comment (tseaver) [inline setup.py]: Why are we pulling in `ZSevrver`?  For WebDAV stuff?  If so, we should make it an optional dependency (i.e., an "extra" called `webdav`) and ensure that imports are conditional, tests are skipped, etc. if it is missing.

File: Products/PluggableAuthService/plugins/tests/ChallengeProtocolChooser.txt [hunk 2]
```diff
@@ -145,9 +145,9 @@ And with valid credentials the reqeuest should succeed:
   ... </methodCall>
   ... """ % (folder_name, username, password), handle_errors=True)
   HTTP/1.1 200 OK
+  ...
   Content-Length: 140
   Content-Type: text/xml
-  <BLANKLINE>
```
Comment (tseaver) [inline Products/PluggableAuthService/plugins/tests/ChallengeProtocolChooser.txt]: This doesn't look correct:  there *should* be a blank line between the headers and the payload.

File: Products/PluggableAuthService/plugins/tests/ChallengeProtocolChooser.txt [hunk 3]
```diff
@@ -181,19 +182,19 @@ normal (eg: browser) request:
   >>> print http(r"""
   ... GET /%s/test_script HTTP/1.1
   ... """ % (folder_name), handle_errors=True)
-  HTTP/1.1 302 Moved Temporarily...
+  HTTP/1.1 302 Found...
```
Comment (tseaver) [inline Products/PluggableAuthService/plugins/tests/ChallengeProtocolChooser.txt]: Wouldn't it be better to use the ellipsis to allow both messages?

File: setup.py [hunk 4]
```diff
@@ -49,7 +49,9 @@ def _package_doc(name):
           'setuptools',
           'Products.PluginRegistry',
           'Products.GenericSetup',
-          'Zope2 >= 2.12',
+          'Products.StandardCacheManagers',
+          'ZServer',
```
Comment (tseaver) [inline setup.py]: Why pull in `ZServer` as an unconditional dependency?  If it is for WebDAV stuff, then we should add a `webdav` extra and ensure that imports are conditional, tests are skipped, etc. if it is missing.

File: setup.py (hunk 1 above)
Comment (icemac) [reply setup.py]: You are right, I created #8 to fix this, so this PR does not get too big.

File: Products/PluggableAuthService/plugins/tests/ChallengeProtocolChooser.txt (hunk 3 above)
Comment (icemac) [reply Products/PluggableAuthService/plugins/tests/ChallengeProtocolChooser.txt]: It was fixed in #7 the way you suggested it.

File: setup.py (hunk 4 above)
Comment (icemac) [reply setup.py]: See #8.

**Review by icemac — CHANGES_REQUESTED** [review]
IMHO this PR should be merged after #7 is merged, as it does some of its fixes again.

**Comment by icemac** [comment]
Now after #7 is merged to master this PR could be rebased to the master.

**Review by icemac — APPROVED** [review]

**Comment by icemac** [comment]
@jensens Can I safely delete the branch – as I merged it to master  – or will this break some integration tests?

**Comment by icemac** [comment]
I decided to delete the branch to signal that the integration tests now should use the master branch.

**Pull request merged**